### PR TITLE
feat(locksmith): Add new login mechanism based on EIP 4361

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

This adds ethereum sign to locksmith described in https://github.com/unlock-protocol/unlock/issues/8550

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "graphql": "15.8.0",
     "graphql-tag": "2.12.6",
     "html-template-tag": "4.0.0",
+    "iron-session": "^6.1.2",
```
Comment (julien51) [inline locksmith/package.json:65]:
```suggestion
    "iron-session": "6.1.2",
```

File: locksmith/package.json [hunk 2]
```diff
@@ -80,6 +81,7 @@
     "sequelize-cli": "6.4.1",
     "sequelize-typescript": "1.1.0",
     "setimmediate": "1.0.5",
+    "siwe": "^1.1.6",
```
Comment (julien51) [inline locksmith/package.json:84]:
```suggestion
    "siwe": "1.1.6",
```

File: locksmith/src/utils/session.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  interface IronSessionData {
+    nonce: string | null
+    siwe: SiweMessage | null
+  }
+}
+export const sessionOptions: IronSessionOptions = {
+  cookieName: 'locksmith',
+  password:
+    process.env.SECRET_COOKIE_PASSWORD ??
+    '32-characters-long-secret-cookie-password',
```
Comment (julien51) [inline locksmith/src/utils/session.ts:16]: 👍

File: locksmith/src/controllers/loginController.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      const message = new SiweMessage(request.body.message)
+      const fields = await message.validate(request.body.signature)
+      if (fields.nonce !== request.session.nonce) {
+        logger.info(`${request.session} failed to validate.`)
+        response.status(422).json({
+          message: 'Invalid nonce',
+        })
+      }
+      request.session.siwe = fields
+      await request.session.save()
```
Comment (julien51) [inline locksmith/src/controllers/loginController.ts:22]: Is that an express API?

**Review by julien51 — APPROVED** [review]
Let's really try to keep track of why the `base` config for jest does not work... but in the interest of time I'll aprove so we can make progress/

File: locksmith/src/controllers/loginController.ts (hunk 4 above)
Comment (searchableguy) [reply locksmith/src/controllers/loginController.ts:22]: no, that is attached by the sessionMiddleware from iron.

**Pull request merged**